### PR TITLE
SCRUM-2019 add parent IDs for each disease term

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

File: agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java [hunk 1]
```diff
@@ -123,6 +116,17 @@ private List<GeneDiseaseAnnotationDocument> createGeneDiseaseAnnotationDocuments
 		return ret;
 	}
 
+	private static Map<String, Set<String>> closureMap = null;
+
+	// Get Closure for disease ontology
+	private static Map<String, Set<String>> getClosure() {
+		if (closureMap == null) {
+			DiseaseRepository diseaseRepository = new DiseaseRepository();
```
Comment (oblodgett) [inline agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java:124]: This DiseaseRepository should be opened as part of the main class and not on each call to getClosure.

File: agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java [hunk 2]
```diff
@@ -123,6 +116,17 @@ private List<GeneDiseaseAnnotationDocument> createGeneDiseaseAnnotationDocuments
 		return ret;
 	}
 
+	private static Map<String, Set<String>> closureMap = null;
```
Comment (oblodgett) [inline agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java:119]: These variables should be defined at the top of the class with the others.

File: agr_java_core/src/main/java/org/alliancegenome/api/entity/DiseaseAnnotationDocument.java [hunk 3]
```diff
@@ -27,6 +27,7 @@ public class DiseaseAnnotationDocument extends SearchableItemDocument {
 	private Reference singleReference;
 	private Set<Reference> references;
 
+	Set<String> parentSlimIDs;
```
Comment (oblodgett) [inline agr_java_core/src/main/java/org/alliancegenome/api/entity/DiseaseAnnotationDocument.java:30]: should be private?

File: agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java (hunk 1 above)
Comment (cmpich) [reply agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java]: This method is only executed once until the map is initialized. Moved it anyway.

File: agr_java_core/src/main/java/org/alliancegenome/api/entity/DiseaseAnnotationDocument.java (hunk 3 above)
Comment (cmpich) [reply agr_java_core/src/main/java/org/alliancegenome/api/entity/DiseaseAnnotationDocument.java]: Yes.

**Review by oblodgett — APPROVED** [review]

**Pull request merged**